tv_apps/dashboard/js/digital_clock.js

```diff
@@ -46,15 +40,17 @@
      * @param  {Function} refresh A callback to run on every tick.
      */
     _start: function (refresh) {
-      var now = new Date();
+      var self = this;
+
+      function loop () {
+        var now = new Date();
 
-      this.timeoutId = setTimeout(function () {
-        refresh(new Date());
+        refresh(now);
+
+        self.timeoutId = setTimeout(loop, 1000 - now.getMilliseconds());
```

Comment:
how about `this.timeoutId = setTimeout(_start.bind(this), 1000 - now.getMilliseconds());`
We won't need to nest a function here then.